test/raven.client.js

```diff
@@ -382,6 +382,98 @@ describe('raven.Client', function () {
       });
     });
 
+    it('should respect shouldSendCallback', function (done) {
+      client = new raven.Client(dsn, {
+        shouldSendCallback: function (data) {
+          return false;
+        }
+      });
+
+      // neither of these should fire, so report err to done if they do
+      client.on('logged', done);
+      client.on('error', done);
+
+      client.process({
+        message: 'test'
+      }, function (err, eventId) {
+        setTimeout(done, 10);
+      });
+    });
+
+    it('should pass original shouldSendCallback to newer shouldSendCallback', function (done) {
+      var cb1 = function (data) {
+        return false;
+      };
+
+      var cb2 = function (data, original) {
+        original.should.equal(cb1);
+        return original(data);
+      };
+
+      var cb3 = function (data, original) {
+        return original(data);
+      };
+
+      client = new raven.Client(dsn, {
+        shouldSendCallback: cb1,
+      });
+
+      client.setShouldSendCallback(cb2);
+      client.setShouldSendCallback(cb3);
+
+      // neither of these should fire, so report err to done if they do
+      client.on('logged', done);
+      client.on('error', done);
+
+      client.process({
+        message: 'test'
+      }, function (err, eventId) {
+        setTimeout(done, 10);
+      });
+    });
+
+    it('should pass original dataCallback to newer dataCallback', function (done) {
+      var scope = nock('https://app.getsentry.com')
+        .filteringRequestBody(/.*/, '*')
+        .post('/api/269/store/', '*')
+        .reply(200, function (uri, body) {
+          zlib.inflate(new Buffer(body, 'base64'), function (err, dec) {
+            if (err) return done(err);
+            var msg = JSON.parse(dec.toString());
+            msg.extra.foo.should.equal('bar');
```

Comment:
Caught me slippin again. I'm actually not sure, could go either way depending on implementation details. Switched to `inflateSync` to be more obviously sequenced.